core/vm/actor/builtin/types/miner/deadline.hpp

```diff
@@ -85,19 +85,19 @@ namespace fc::vm::actor::builtin::types::miner {
      * will be moved to PoStSubmissionsSnapshot. WindowPoSt proofs verified
      * on-chain do not appear in this AMT.
      */
-    adt::Array<WindowedPoSt> optimistic_post_submissions;
+    adt::Array<WindowedPoSt, 2> optimistic_post_submissions;
 
     /**
      * Snapshot of partition state at the end of the previous challenge window
      * for this deadline.
      */
-    CID partitions_snapshot;
+    decltype(partitions) partitions_snapshot;
```

Comment:
`decltype(partitions)` = `adt::Array<Universal<Partition>, 3>` = `CID + Ipld`
But we want to use it just as CID, as I understand.
Ipld for snapshots is not load by visitor and if anybody missed id and call something related to Ipld from snapshots, then we have an error

Reply:
No, `DisputeWindowedPoSt` uses them as `Array`.

Reply:
In this case we should load Ipld